os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +255,14 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+		mpu_configure_app_regs(&tcb->mpu_regs[STACKGUARD_MPU_NUM * REG_MAX], STACKGUARD_MPU_NUM,
+			(uint32_t)tcb->stack_alloc_ptr, MPU_REGION_MINIMUM_SIZE, true, false);
+		/* Adjust the adj_stack_size after guard_size calculationn */
+		tcb->adj_stack_size = tcb->adj_stack_size - MPU_REGION_MINIMUM_SIZE;
```

Comment:
You allocate `stack_size` user requested. If we need stack guard, don't we need to add the guard size?
I mean
```
tcb->stack_alloc_ptr = kmm_....(stack_size + MPU_REGION_MINIMUM_SIZE);
tcb->adj_stack_size = tcb->adj_stack_size - MPU_REGION_MINIMUM_SIZE;
```

Reply:
Whatever we support, the most important is we should guarantee user request. Even user want `stack_size`, we always return `stack_size - MPU_REGION_MINIMUM_SIZE`. I don't think this is good.

Reply:
yes, modified and allocated user requested size